### PR TITLE
[#120869561] Filling in Sanger Sequencing Well Plate

### PR DESCRIPTION
**Pull request by jhanggi**
This pulls in Vue for filling in the well plates. 

You can now add submissions to a well plate in odd-rows first order.

Still lacking from this story:
* Multiple well-plates within a batch
* Colors
* Removing a submission from a plate

Special thanks to @theotherzach for his help teaching me how Vue works.

### CODE REVIEW

File: vendor/engines/sanger_sequencing/app/assets/javascripts/sanger_sequencing/vue_well_plate_app.coffee [hunk 1]
```diff
@@ -0,0 +1,30 @@
+window.vue_sanger_sequencing_well_plate_app = {
+  props: ["submissions"]
+  data: ->
+    plate: SangerSequencing.WellPlateBuilder.rows()
+    builder: new SangerSequencing.WellPlateBuilder
+
+  ready: ->
+    vueBus.$on "submission-added", @addSubmission
+    new AjaxModal(".js--modal", ".js--submissionModal")
+    console.debug new SangerSequencing.WellPlateBuilder()
```
Comment (drench) [inline vendor/engines/sanger_sequencing/app/assets/javascripts/sanger_sequencing/vue_well_plate_app.coffee]: Presuming the `console.debug` is also temporary

**Comment by drench** [comment]
It works nicely for me so far, and it's really cool to see vue.js in the project. The CodeClimate complaints look legit but small. :+1: with those cleanups.

**Comment by drench** [comment]
:+1:

**Comment by theotherzach** [comment]
I'm late, but 👍

**Pull request merged**